src/main/scala/chisel3/util/experimental/BoringUtils.scala

```diff
@@ -193,4 +204,54 @@ object BoringUtils {
     sinks.foreach(addSink(_, genName, true, true))
     genName
   }
+
+  /** Access a source [[Data]] that may or may not be in the current module.  If
+    * this is in a child module, then create ports to allow access the
+    * requrested source.
+    */
+  def bore(source: Data)(implicit si: SourceInfo): Data = {
+    experimental.requireIsHardware(source)
+
+    val thisModule = Builder.currentModule.get
+
+    source.topBindingOpt match {
+      case None =>
+        Builder.error(s"Cannot bore from $source")
+        Wire(chiselTypeOf(source)) //return empty wire to continue elaboration to collect other errors
+      case Some(internal.CrossModuleBinding) =>
+        Builder.error(s"Cannot bore across a Definition/Instance boundary: ${thisModule} cannot access ${source}")
+        Wire(chiselTypeOf(source)) //return empty wire to continue elaboration to collect other errors
+      case _ => // Actually bore
+
+        var rhs = source
+        var module = source.topBinding.location.get
+        while (module != thisModule) {
+          if (!module.isClosed) {
+            Builder.error(s"Can only bore into modules that have finished building, but $source is in an open module")
+            module = thisModule // exit early
+          } else {
+
+            /** Create a port. */
+            val bore = SecretIO(chiselTypeOf(source), module)
+            module.addSecretIo(bore)
+            module.asInstanceOf[RawModule].secretConnection(bore, rhs)
+
+            rhs = bore
```

Comment:
stupid question -- is this actually doing a connection through the bored IOs? where does that happen?

In other words, its not ust a secret IO that we need, its also a secret connection